modules/ding_place2book/ding_place2book.module

```diff
@@ -536,29 +540,37 @@ function ding_place2book_get_event_state($event_maker_id, $event_id) {
             'end' => strtotime($price->sale_end_at),
           ];
         }
+
         // Some events, for example those migrated from the old API, will have
         // sales period on the event and not on the individual tickets.
-        if (isset($event->sale_open_at) && isset($event->sale_close_at)) {
+        if (isset($event->sale_open_at) && isset($event->sale_close_at) && !$sale_periods) {
```

Comment:
I'm wondering about the `!$sale_periods` condition. 

I don't think it's needed and we don't know if an event can both have sale period on the event while also having sale periods on individual tickets. In any case, I think it will best to just remove it.